### PR TITLE
Fix React whitespace emit 

### LINKED ISSUE #4376: JSX: Whitespace missing in emit for `<something/> foo`
Consider the following:

```ts
<div>word <code>code</code> word</div>
```
it emits : 

```js
React.createElement("div", null, "word ", React.createElement("code", null, "code"), "word");
```

[Should emit](https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&playground=false&code=%3Cdiv%3Eword%20%3Ccode%3Ecode%3C%2Fcode%3E%20word%3C%2Fdiv%3E):

```js
React.createElement("div", null, "word ", React.createElement("code", null, "code"), " word");
```

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #4376

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -6463,7 +6463,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
 
             function trimReactWhitespace(node: JsxText): string {
                 let result: string = undefined;
-                let text = getTextOfNode(node);
+                let text = getTextOfNode(node, true);
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: What is going to be `true` here?

File: src/compiler/emitter.ts [hunk 2]
```diff
@@ -6498,7 +6498,7 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 switch (compilerOptions.jsx) {
                     case JsxEmit.React:
                         let text = trimReactWhitespace(node);
-                        if (text.length === 0) {
+                        if (text === undefined || text.length === 0) {
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: Since `""` if falsy, you could also just return `text || undefined`, but this might just be clearer anyway.

Comment (yuit) [inline src/compiler/emitter.ts]: how can this be undefined?

**Comment by yuit** [comment]
lgtm

**Pull request merged**